store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -6147,10 +6152,21 @@ private void checkPasswordStrength(String password, Account acct, Cos cos, ZMuta
                     new Argument(Provisioning.A_zimbraPasswordMaxLength, maxLength, Argument.Type.NUM));
         }
 
-        if (LC.allow_username_within_password.booleanValue()
-                && StringUtils.containsIgnoreCase(password, acct.getUCUsername())) {
-            throw AccountServiceException.INVALID_PASSWORD("password contains username",
-                    new Argument("zimbraPasswordAllowUsername", "", Argument.Type.STR));
+        // userName is passed from createAccount() and else condition is invoked 
+        // when called from other Api's Changepassword and Resetpassword Api
+        if (userName != null) {
```

Comment:
add a util function `checkPasswordHasUsername` with string username & password as param to check if password contains username and throw AccountServiceException exception from there itself, instead of repeat code here in if else

Reply:
done